### PR TITLE
Fix: Get Version string from the releaseRegistry

### PR DESCRIPTION
**Pull request by Chickensoupwithrice**
- **Hackathon attempt to get latest version into docs**
- **POC for Versions being updated from the ReleaseRegistry**
- **Remove Versions.tsx, it's no longer required**
- **Update all instances of "v5.6.2535" to be {CURRENT_VERSION}**
- **remove flake.nix whoops**

Our version strings are always out of date because we don't have any automation to update them. This PR resolves that problem by reaching out to the ReleaseRegistry at build time to get the latest version string. It then replaces that string in any code block using the PreCodeBlock component (which was previously broken and is now fixed). 
It also adds an exported const `CURRENT_VERSION_STRING` which can be used outside of Code Blocks and is similarly populated at build time. 

The final stage of this PR is automation that kicks off a build and releases the docs from the release pipeline itself so nobody has to go in and do that manually. 

## Pull Request approval

Although pull request approval is not enforced for this repository in order to reduce friction, merging without a review will generate a ticket for the docs team to review your changes. So if possible, have your pull request approved before merging.

### CODE REVIEW

**Comment by MaedahBatool** [comment]
Hey @Chickensoupwithrice the Vercel build is failing. Looks like a `type` issue. You can run `pnpm run build` to view build errors locally.

**Comment by Chickensoupwithrice** [comment]
Yes `CURRENT_VERSION_STRING` currently doesn't work, I still need to fix that. Converted to draft to avoid confusion for the time being.

**Comment by Chickensoupwithrice** [comment]
We still need automation to build the docs again on new release, which I think is a better solution than moving the current_version to client side

**Comment by jdpleiness** [comment]
> We still need automation to build the docs again on new release, which I think is a better solution than moving the current_version to client side 

They get rebuilt during our release process when we merge the changelog anyhow, I believe. While I agree automation is the way to go, we can probably split that into another PR down the line. wdyt?

**Review by jdpleiness — APPROVED** [review]
🚀 🙏

File: src/components/MdxComponents.tsx [hunk 1]
```diff
@@ -9,63 +9,63 @@ import { LinkCard, LinkCards } from './mdx/LinkCards';
...
-			const isCodeBlock = props.style?.display === 'grid';
-			// Only apply styles to inline code
-			const codeClasses = isCodeBlock
-				? 'bg-transparent'
-				: 'border font-medium bg-slate-100 dark:bg-slate-900 px-1 py-0.5 rounded-md border-slate-300 dark:border-slate-700';
+  return {
+    FeatureParity,
+    ResourceEstimator,
+    AWSOneClickLaunchForm,
+    Accordion,
```
Comment (DaedalusG) [inline src/components/MdxComponents.tsx:21]: Indentations 🤦🏼‍♂️

**Comment by Chickensoupwithrice** [comment]
> They get rebuilt during our release process when we merge the changelog anyhow, I believe. While I agree automation is the way to go, we can probably split that into another PR down the line

Yes absolutely.

File: src/components/MdxComponents.tsx [hunk 2]
```diff
@@ -9,63 +9,63 @@ import { LinkCard, LinkCards } from './mdx/LinkCards';
...
+    ProductCards,
+    ProductCard: (props: React.ComponentProps<typeof ProductCard>) => (
+      <ProductCard {...props} version={version} />
+    ),
+    code: (props: any) => {
+      const isCodeBlock = props.style?.display === 'grid';
+      // Only apply styles to inline code
+      const codeClasses = isCodeBlock
+        ? 'bg-transparent'
+        : 'border font-medium bg-slate-100 dark:bg-slate-900 px-1 py-0.5 rounded-md border-slate-300 dark:border-slate-700';
```
Comment (Chickensoupwithrice) [inline src/components/MdxComponents.tsx:45]: We can remove these lines now, they exist in the PreCode component

File: src/components/PreCodeBlock.tsx [hunk 3]
```diff
@@ -5,23 +5,113 @@ interface PreProps extends DetailedHTMLProps<HTMLAttributes<HTMLPreElement>, HTM
...
+    React.Children.forEach(children, (child) => {
+      if (React.isValidElement(child) && child.props.children) {
+        codeContent += child.props.children
+      }
+    })
+  }
+
+  const version = await getLatestVersion()
+  const version_no_v = trimPrefix(version, "v")
+  codeContent = codeContent.replace(/{CURRENT_VERSION}/g, version)
```
Comment (DaedalusG) [inline src/components/PreCodeBlock.tsx:75]: Nice so no special release just whenever the docsite builds this parent component it'll get the info from the release registry 🚀

**Review by DaedalusG — APPROVED** [review]

File: src/components/PreCodeBlock.tsx [hunk 4]
```diff
@@ -5,23 +5,113 @@ interface PreProps extends DetailedHTMLProps<HTMLAttributes<HTMLPreElement>, HTM
...
+  let codeContent = props?.raw || ''
+
+  // If raw prop is not available, try to extract content from children
+  if (!codeContent && children) {
+    React.Children.forEach(children, (child) => {
+      if (React.isValidElement(child) && child.props.children) {
+        codeContent += child.props.children
+      }
+    })
+  }
```
Comment (Chickensoupwithrice) [inline src/components/PreCodeBlock.tsx:104]: We only use `children` here

Comment (Chickensoupwithrice) [reply src/components/PreCodeBlock.tsx:104]: (aka can remove this bit)

File: src/components/MdxComponents.tsx [hunk 5]
```diff
@@ -9,63 +9,63 @@ import { LinkCard, LinkCards } from './mdx/LinkCards';
...
```
Comment (Chickensoupwithrice) [inline src/components/MdxComponents.tsx:44]: This lives in the `PreCode` component now, can also be removed

File: src/components/MdxComponents.tsx (hunk 1 above)
Comment (Chickensoupwithrice) [reply src/components/MdxComponents.tsx:21]: I blame `vstls`

File: src/components/PreCodeBlock.tsx (hunk 3 above)
Comment (Chickensoupwithrice) [reply src/components/PreCodeBlock.tsx:75]: Yessir! We have 2 components, and 2 replacements in each of them. One for inline code blocks, and one for entire <pre> code blocks. Then there's one replacement for the version with the `v`, and one for without the `v`

**Comment by DaedalusG** [comment]
This change is probably worth a note somewhere in the doc writing handbook, inquiring about that here: https://sourcegraph.slack.com/archives/C01DXLN3D0T/p1727460257629219

**Pull request merged**